lib/src/input/hekla.rs

```diff
@@ -0,0 +1,91 @@
+use core::fmt::Debug;
+
+use alloy_sol_types::sol;
+use serde::{Deserialize, Serialize};
+
+sol! {
```

Comment:
One cool thing I've seen being used is instead of copy pasting the relevant code here, is to simply give it the abi json output of the solidity compiler:
```
sol!(RollupContract, "TaikoL1.json");
use RollupContract::{BlockProposed, RollupContractCalls, RollupContractEvents};
```

Reply:
nice, will try!